scripts/un/sdg/geography.py

```diff
@@ -0,0 +1,325 @@
+            if not subject or not object:
+                continue
+            un2sdg[subject] = object
+            sdg2un[object] = subject
+    return un2sdg, sdg2un
+
+
+def get_un2dc(file):
+    '''Produces map of UN code -> curated (dcid, DC type, DC name).
+
+    Args:
+        file: Input file path.
+
+    Returns:
+        Map of UN code -> curated (dcid, DC type, DC name).
+    '''
+    un2dc = {}
+    with open(file) as f:
+        reader = csv.DictReader(f)
+        for row in reader:
+
+            # Skip unmapped places.
+            if row['unDataCode'] == 'x':
+                continue
+
+            # Add missing type for NorthernEurope.
+            if row['dcid'] == 'NorthernEurope':
+                type = 'UNGeoRegion'
+
+            else:
+                type = json.loads(row['typeOf'].replace("'", '"'))[0]['dcid']
+            un2dc[row['unDataCode']] = (row['dcid'], type, row['dc_name'])
+    return un2dc
+
+
+def should_include_containment(s_type, s_dcid, o_type, o_dcid):
+    '''Returns whether triple should be included in containment.
+
+    Args: 
+        s_type: Type of triple subject.
+        s_dcid: Dcid of triple subject.
+        o_type: Type of triple object.
+        o_dcid: Type of triple object.
+
+    Returns:
+        Whether triple should be included in containment.
+    '''
+    if (s_type == 'GeoRegion' or s_type == 'UNGeoRegion') and o_dcid == 'Earth':
+        return True
+    elif (s_type == 'GeoRegion' or
+          s_type == 'UNGeoRegion') and o_type == 'Continent':
+        return True
+    elif (s_type == 'GeoRegion' or
+          s_type == 'UNGeoRegion') and (o_type == 'GeoRegion' or
+                                        o_type == 'UNGeoRegion'):
+        return True
+    elif s_type == 'Country' and (o_type == 'GeoRegion' or
+                                  o_type == 'UNGeoRegion'):
+        return True
+    elif s_type == 'SamplingStation' and o_type == 'Country':
+        return True
+    elif s_type == 'City' and s_dcid.startswith(
+            'undata-geo') and o_type == 'Country':
+        return True
+    return False
+
+
+def write_un_places(input_geos, output, sdg2type, un2sdg, un2dc):
+    '''Writes UN places to output and computes new places.
+
+    Args: 
+        input_geos: Path to input UN geography file.
+        output: Path to output file.
+        sdg2type: Map of SDG code -> SDG type.
+        un2sdg: Map of UN code -> SDG code.
+        un2dc: Map of UN code -> curated (dcid, DC type, DC name).
+
+    Returns:
+        - Map of UN code -> generated (dcid, DC type, DC name).
+        - Set of (dcid, type) for new places.
+    '''
+    un2dc2 = {}
+    new_subjects = set()
+    with open(input_geos) as f_in:
+        with open(output, 'w') as f_out:
+            reader = csv.DictReader(f_in)
+            for row in reader:
+                subject = row['subject_id']
+                if subject in un2dc:
+                    dcid = un2dc[subject][0]
```

Comment:
nit: dcid, type, name = un2dc[subject] ?

Reply:
switched to use class